packages/components/src/number-control/index.tsx

```diff
@@ -120,7 +123,6 @@ function UnforwardedNumberControl(
 				type === inputControlActionTypes.PRESS_UP ||
 				type === inputControlActionTypes.PRESS_DOWN
 			) {
-				// @ts-expect-error TODO: Resolve discrepancy between `value` types in InputControl based components
```

Comment:
Just to clarify this only-slightly related change:

The discrepancy here was caused by `constrainValue()` either returning a number or a string and in this PR I've updated this function so that it only returns a string (see [line 73](https://github.com/WordPress/gutenberg/pull/52902/files#diff-1795d70ae6f25ef5ee79fa75d215a18e87de7d7bdc577b294376838325d983a8R73)), which is why these `@ts-expect-error`s are no longer needed. I believe a string is all that is needed, as input values are always strings.